crates/primitives/src/bytecode.rs

```diff
@@ -66,10 +66,35 @@ impl Bytecode {
         matches!(self, Self::Eof(_))
     }
 
+    /// Create a new legacy raw [`Bytecode`].
+    pub fn new_legacy(bytecode: Bytes) -> Self {
+        Self::LegacyRaw(bytecode)
+    }
+
+    /// Checks if the bytecode starts with the EOF prefix followed by the correct version.
+    #[inline]
+    pub fn is_eof_format(bytes: &Bytes) -> bool {
+        if bytes.len() >= 3 {
+            let prefix = u16::from_be_bytes([bytes[0], bytes[1]]);
+            let version = bytes[2];
+
+            prefix == EOF_PREFIX && version == EOF_VERSION
+        } else {
```

Comment:
This can be missleading in tx create case. For EOF only first two bytes should be checked.

You can do this as:
`.get(0..2).filter(|&t| t == &EOF_MAGIC_BYTES).is_some()`